src/cli/cmd_build_and_check.rs

```diff
@@ -184,7 +205,7 @@ fn produce_all_asts<B: Backend>(path: &PathBuf) -> miette::Result<(Sources, Type
     let mut tast = TypeChecker::new();
 
     // adding stdlib
-    add_stdlib(&mut sources, &mut tast, node_id)?;
+    add_stdlib(&mut sources, &mut tast, node_id, server_mode)?;
```

Comment:
maybe it is better to create server_mode fields in nast/tast/mast structs? so we don't have to add `server_mode` as an argument everytime we call the functions?

Reply:
yeah I think we should have some sort of super `Options` struct so that we can manage all of these arguments that we pass in different places, and have it have a `Default` implementation as well as setters